docker/Dockerfile

```diff
@@ -0,0 +1,47 @@
+ARG UBUNTU_VERSION=18.04
+FROM ubuntu:${UBUNTU_VERSION}
+
+# Use bash as the default shell
+RUN ln -snf /bin/bash /bin/sh
+
+# Update core packages
+RUN apt-get -y update
+RUN apt-get -y upgrade
```

Comment:
Unfortunately, it's against best practices to run `apt-get -y upgrade` or `apt-get -y dist-upgrade` from within a docker container.  From the dockerfile documentation [1]:

```
Avoid RUN apt-get upgrade and dist-upgrade, as many of the
“essential” packages from the parent images cannot upgrade
inside an unprivileged container.

If a package contained in the parent image is out-of-date,
contact its maintainers. If you know there is a particular package,
foo, that needs to be updated, use apt-get install -y foo to update automatically.
```

[1] https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run